tests/common.py

```diff
@@ -403,6 +403,9 @@ def check_env_recovery_with_subset_spaces(
         observation_space_subset (gym.spaces.Space): desired subset observation space
 
     """
+    if dataset.env_spec is None:
+        raise ValueError("Recovering environment is not possible when env_spec is None")
```

Comment:
this it is not necessary I believe, as `recover_environment` will throw an error itself if env_spec is None

Reply:
removed the redundant raise error